src/api.py

```diff
@@ -0,0 +1,18 @@
+import requests 
+import hashlib
+def fetch(url):
+    con = requests.head(url)
+    r = requests.get(url)
```

Comment:
What if there is an error while fetching?

Reply:
Okay I should have used try and catch block

Reply:
catching every error is probably not the way.You want to catch some of them and throw a well defined exception instead

Reply:
Ohhkay now I am clear, thanks for explaining